packages/contract/lib/promievent.js

```diff
@@ -1,36 +1,107 @@
+const debug = require("debug")("contract:promievent");
+const DebugUtils = require("@truffle/debug-utils");
 const Web3PromiEvent = require("web3-core-promievent");
 
-function PromiEvent() {
-  const { resolve, reject, eventEmitter } = Web3PromiEvent.apply(
-    null,
-    arguments
-  );
+function PromiEvent(justPromise, bugger) {
```

Comment:
You should just use new class syntax, don't worry about the style consistency for that sorta thing. We must have forward progress!

Reply:
Hm, turns out that broke things for some reason.  Leaving this alone for now actually.